packages/hierarchy-browser/src/hooks/useUpdatedCommunityProvider.ts

```diff
@@ -0,0 +1,35 @@
+/*!
+ * Copyright (c) Microsoft. All rights reserved.
+ * Licensed under the MIT license. See LICENSE file in the project.
+ */
+import { useMemo, useEffect } from 'react'
+import { CommunityDataProvider } from '../common/dataProviders'
+import { HierarchyDataProvider } from '../common/dataProviders/HierachyDataProvider'
+import { ICommunityDetail, ILoadNeighborCommunities } from '../types'
+
+export function useUpdatedCommunityProvider(
+	hierachyDataProvider: HierarchyDataProvider,
+	community: ICommunityDetail,
+	level: number,
+	neighborCallback?: ILoadNeighborCommunities,
+	dataProvider?: CommunityDataProvider,
+) {
+	useEffect(() => {
+		if (dataProvider) {
+			dataProvider.updateCommunityData(community)
+			dataProvider.updateHierarchyDataProvider(hierachyDataProvider)
+		}
+	}, [community, hierachyDataProvider])
```

Comment:
dataProvider dep? Does the linter pass here? We should get warnings about incorrect hook deps